### DISCUSSION TITLE
Syntax highlighting recommendations for Native WC

### DISCUSSION THREAD

**Discussion by MichaelTrilford-Airwallex** [mdn/mdn-community#555] [Content]
For Shadow.Root docs, can we add some helpful guidelines on MDN of how we can enable syntax highlighting for native web components?

This is one area that I struggle to find any information on, and I would love to learn more about achieving that.

https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot/innerHTML

`shadow.innerHTML = '<strong>This element should be more important!</strong>';`

**Comment by Rumyra** [comment]
Just to clarify @MichaelTrilford-Airwallex you would like guidelines for highlighting code displayed _inside_ components?

**Reply by MichaelTrilford-Airwallex** [reply]
Hello Ruth,

Here's an example:

<img width="475" alt="Screen Shot 2021-07-20 at 6 02 21 pm" src="https://user-images.githubusercontent.com/75596057/126284420-52993600-b8b0-4bbe-9d84-b8523cc5724b.png">

See how the colour is orange and there is no formatting? I'm struggling to find resources to enable correct syntax highlighting.

Whenever I'm stuck, I find myself on your website. Whether it is the right place or not. Maybe the MDN Web docs could be a good place to expose other resources.

<img width="417" alt="Screen Shot 2021-07-20 at 6 08 52 pm" src="https://user-images.githubusercontent.com/75596057/126285287-231e38a2-3c80-410c-8c45-f97476b6f121.png">

**Reply by Ryuno-Ki** [reply]
I've seen Styled Components sporting syntax highlighting for VS Code. Perhaps you can study its implementation and check, whether it can be applied to this case as well?

Might be, that you need to turn the template literal into a tagged template, so it's easier for an IDE to separate this use from a regular template literal.

**Reply by MichaelTrilford-Airwallex** [reply]
Thanks mate. I have seen Web Components that put html in front, and that works.

E.g.
`html’…’` – I had to use regular ticks for this example

But usually you need to spin up a certain framework.

I guess some resources around best practices / boilerplate examples of structuring a native web component would be very beneficial for people wanting to ‘buy-in’ to Web Components.
